### PR TITLE
share: remember original/compressed setting in local DB

### PR DESCRIPTION
**Pull request by songgao**
depends on https://github.com/keybase/client/pull/23954

### CODE REVIEW

File: protocol/avdl/keybase1/incoming-share.avdl [hunk 1]
```diff
@@ -24,4 +24,11 @@ protocol incomingShare {
   }
 
   array<IncomingShareItem> getIncomingShareItems();
+
+  enum IncomingShareCompressPreference {
+    ORIGINAL_0,
+    COMPRESSED_1
```
Comment (joshblum) [inline protocol/avdl/keybase1/incoming-share.avdl:30]: can this be a struct that also has the chat/kbfs preference instead?

File: go/libkb/db.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ const (
 	DBTeamChain         = 0x10
 	DBUserPlusAllKeysV1 = 0x19
 
+	DBIncomingSharePreference        = 0xa4
```
Comment (joshblum) [inline go/libkb/db.go:24]: can you addthis to `IsPermDbKey` below/

Comment (songgao) [reply go/libkb/db.go:24]: Ahh good call. I actuallly saw that first and thought this needed to be there but forgot. Thanks

File: protocol/avdl/keybase1/incoming-share.avdl (hunk 1 above)
Comment (songgao) [reply protocol/avdl/keybase1/incoming-share.avdl:30]: It always goes to chat now, and there’s a footer button allowing you to go to kbfs. I can’t think of another setting right now, but if we do in the future perhaps we can use a different key with the same type in the DB key?

Comment (joshblum) [reply protocol/avdl/keybase1/incoming-share.avdl:30]: we'd have to migrate old entries, i think in general it's better to have a struct so we can add fields at our leisure. \

Comment (songgao) [reply protocol/avdl/keybase1/incoming-share.avdl:30]: hmm not sure we have to migrate in that case. Currently the key for this compress preference is:

```
libkb.DbKey{
	Typ: libkb.DBIncomingSharePreference,
	Key: "v0-compress-preference",
}
```

If we need more, we can just use 

```
libkb.DbKey{
	Typ: libkb.DBIncomingSharePreference,
	Key: "v0-other-preference",
}
```

I like that this makes them independent from each other. Other wise when we set a value we have to read the old struct out first, change the field we want to change, then write it back. Either that, or deal with the struct in TS too.

**Comment by joshblum** [comment]
That’s a disk read per pref then, think a struct  is easiest

**Comment by songgao** [comment]
> That’s a disk read per pref then, think a struct is easiest

That's assuming we plumb the struct all the way to frontend, which makes sense if they are always used together. Otherwise if we use Go to handle changing one field in the struct, it'd need to read the struct out, then write in. Depends on what the future fields will be I guess. but I don't feel strongly about this either way so I'll just change it into a struct.

**Comment by songgao** [comment]
Actually I think you're right @joshblum ! After I put the struct changes in this does look better

**Pull request merged**